### ISSUE TITLE
NIP 33 - Too many events being deleted

### ISSUE THREAD

**Issue by hendore** [scsibug/nostr-rs-relay#59]
When publishing an event within `30000 <= n < 40000`, it appears all previous events of the same kind are deleted regardless of what the value of the `d` tag is.

```js
// test event 1
{
  "kind": 30000,
  "tags": [
    [ "d", "test" ],
    [ "foo", "bar" ]
  ],
  ...
}

// test event 2
{
  "kind": 30000,
  "tags": [
    [ "d", "another-test" ],
    [ "foo", "baz" ]
  ],
  ...
}
```

When publishing the 2nd test event, the relay appears to be deleting the first test event even though they have different identifier tags. It looks like all events with the same kind and author are removed and the identifier tag is ignored. Is this correct behaviour?

**Comment by scsibug** [comment]
Nope, definitely a bug, i'm not preserving other events with different 'd' tags.  Will investigate and fix; thanks!

**Comment by hendore** [comment]
@scsibug I would have loved to help out with a PR but I have zero rust knowledge. That said, thank you so much for your effort, running the relay locally in docker has been brilliant 👏🏼

**Comment by scsibug** [comment]
Fixed in 8004ea9, will be in release 0.8.2.

Thanks again for pointing this out!
